src/structure.js

```diff
@@ -68,6 +77,54 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path) {
+  var self = this, pathId = pathString(path);
+  var listenerNs = self._pathListeners[pathId];
```

Comment:
I'm wondering there should be another dimension to `self._pathListeners[pathId]`. One for reference cursors.

Example:
`self._pathListeners[pathId][referenceKey][funIdx]`